frame/curve-amm/src/tests.rs

```diff
@@ -119,111 +122,65 @@ fn get_y_j_greater_than_n() {
-		assert_ok!(Tokens::mint_into(MockCurrencyId::USDT, &ALICE, 200000));
-		assert_eq!(Tokens::balance(MockCurrencyId::USDT, &ALICE), 200000);
-		assert_eq!(Tokens::balance(MockCurrencyId::BTC, &ALICE), 0);
-		assert_ok!(Tokens::mint_into(MockCurrencyId::BTC, &ALICE, 20));
-		assert_eq!(Tokens::balance(MockCurrencyId::BTC, &ALICE), 20);
-		assert_eq!(Tokens::balance(MockCurrencyId::USDT, &BOB), 0);
-		assert_ok!(Tokens::mint_into(MockCurrencyId::USDT, &BOB, 200000));
-		assert_eq!(Tokens::balance(MockCurrencyId::USDT, &BOB), 200000);
-		assert_eq!(Tokens::balance(MockCurrencyId::BTC, &BOB), 0);
-		assert_ok!(Tokens::mint_into(MockCurrencyId::BTC, &BOB, 20));
-		assert_eq!(Tokens::balance(MockCurrencyId::BTC, &BOB), 20);
-		let p = CurveAmm::create_pool(&ALICE, assets, amp_coeff, fee, admin_fee);
-		assert_ok!(&p);
-		let pool_id = p.unwrap();
-		let pool = CurveAmm::pool(pool_id);
-		assert!(pool.is_some());
-		let pool_info = pool.unwrap();
-		// 1 BTC = 65000 USDT
-		let amounts = vec![2u128, 130000u128];
-		assert_ok!(CurveAmm::add_liquidity(&ALICE, pool_id, amounts.clone(), 0u128));
-		let alice_balance = Tokens::balance(pool_info.pool_asset, &ALICE);
-		assert_ne!(alice_balance, 0);
-		assert_eq!(Tokens::balance(MockCurrencyId::USDT, &ALICE), 200000 - 130000);
-		assert_eq!(Tokens::balance(MockCurrencyId::BTC, &ALICE), 20 - 2);
-		let pool = CurveAmm::pool(pool_id);
-		assert!(pool.is_some());
-		let pool_info = pool.unwrap();
-		assert_ok!(CurveAmm::add_liquidity(&BOB, pool_id, amounts.clone(), 0u128));
-		let bob_balance = Tokens::balance(pool_info.pool_asset, &BOB);
-		assert_ne!(bob_balance, 0);
-		assert_eq!(Tokens::balance(MockCurrencyId::USDT, &BOB), 200000 - 130000);
-		assert_eq!(Tokens::balance(MockCurrencyId::BTC, &BOB), 20 - 2);
-		let min_amt = vec![0u128, 0u128];
-		assert_eq!(Tokens::balance(MockCurrencyId::BTC, &CurveAmm::account_id(&pool_id)), 4);
-		assert_eq!(Tokens::balance(MockCurrencyId::USDT, &CurveAmm::account_id(&pool_id)), 260000);
-		assert_ok!(CurveAmm::remove_liquidity(&ALICE, pool_id, alice_balance, min_amt.clone()));
-		assert_eq!(Tokens::balance(pool_info.pool_asset, &ALICE), 0);
-		assert_eq!(Tokens::balance(MockCurrencyId::USDT, &ALICE), 200000);
-		assert_eq!(Tokens::balance(MockCurrencyId::BTC, &ALICE), 20);
-		assert_eq!(Tokens::balance(MockCurrencyId::BTC, &CurveAmm::account_id(&pool_id)), 2);
-		assert_eq!(Tokens::balance(MockCurrencyId::USDT, &CurveAmm::account_id(&pool_id)), 130000);
-		assert_ok!(CurveAmm::remove_liquidity(&BOB, pool_id, bob_balance, min_amt.clone()));
-		assert_eq!(Tokens::balance(pool_info.pool_asset, &BOB), 0);
-		assert_eq!(Tokens::balance(MockCurrencyId::USDT, &BOB), 200000);
-		assert_eq!(Tokens::balance(MockCurrencyId::BTC, &BOB), 20);
-		assert_eq!(Tokens::balance(MockCurrencyId::BTC, &CurveAmm::account_id(&pool_id)), 0);
-		assert_eq!(Tokens::balance(MockCurrencyId::USDT, &CurveAmm::account_id(&pool_id)), 0);
-	});
-}
-
 #[test]
 fn exchange_test() {
```

Comment:
Can we have more than 1 test for exchange, can we have some hardcoded values with slippage for instance?

Reply:
There are more than 1 exchanges in test. I will update test with hardcoded values.